torch/jit/_recursive.py

```diff
@@ -123,6 +125,10 @@ def infer_type(name, item):
     added_names = set()
 
     for name, item in nn_module._parameters.items():
+        if name in user_annotated_ignored_attributes:
```

Comment:
I don't think we need to check `user_annotated_ignored_attributes` for each present attribute, we can just add it all to the concrete type right away, e.g. 
```
concrete_type_builder.add_ignored_attributes(user_annotated_ignored_attributes)
```

The only difference would be that ignored attributes would be added even if they aren't present in the module object, but that won't make a difference in practice I think.

Reply:
But we still need to skip adding attributes if they are ignored, right? So instead of 
```
if name in user_annotated_ignored_attributes:
    concrete_type_builder.add_ignored_attribute(name)
    continue
```
we'd have
```
if name in user_annotated_ignored_attributes:
    continue
```
or 
```
if {existing_condition} and name not in user_annotated_ignored_attributes:
     concrete_type_builder.add_xxx(...)
```